src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -73,6 +101,114 @@ def __init__(self, proto_unix_io):
         self.geneva_value = 3
         self.power_value = 1
 
+    def kick_clicked(self):
+        """
+        If buttons are enabled, sends a Kick command and disables buttons
+
+        Attaches a callback to re-enable buttons after 3 seconds
+        """
+        # if button is enabled
+        if self.buttons_enable:
+            # send kick primitive
+            self.send_command(ChickerCommandMode.KICK)
+            self.disable_kick_chip_button()
+
+            # set and start timer to re-enable kick button after 3 seconds
+            self.start_timer_once(
+                self.enable_kick_chip_button, 3 * MILLISECONDS_PER_SECOND
+            )
+
+    def chip_clicked(self):
+        """
+        If buttons are enabled, sends a Chip command and disables buttons
+
+        Attaches a callback to re-enable buttons after 3 seconds
+        """
+
+        # if button is enabled
+        if self.buttons_enable:
+            # send chip primitive
+            self.send_command(ChickerCommandMode.CHIP)
+            self.disable_kick_chip_button()
+
+            # set and start timer to re-enable chip button after 3 seconds
+            self.start_timer_once(
+                self.enable_kick_chip_button, 3 * MILLISECONDS_PER_SECOND
+            )
+
+    def start_timer_once(self, function, duration):
+        """Starts a QTimer to call the given function once after the given duration
+
+        :param function: function to call after duration
+        :param duration: duration to call function after
+        :returns: None
+
+        """
+
+        timer = QTimer(self)
+        timer.setTimerType(Qt.TimerType.PreciseTimer)
+        timer.timeout.connect(function)
+        timer.setSingleShot(True)
+        timer.start(duration)
+
+    def disable_kick_chip_button(self):
+        """
+        If buttons are enabled, disables them
+        """
+        if self.buttons_enable:
+            self.buttons_enable = False
+
+    def enable_kick_chip_button(self):
+        """
+        If buttons are disabled and buttons should be enabled, enables them
+        """
+        if not self.buttons_enable and self.buttons_should_enable:
+            self.buttons_enable = True
```

Comment:
It seems like `buttons_should_enable` is true if auto kick/chip is selected, and false otherwise. It might be a bit more descriptive if it's renamed to something like `auto_kick_chip_selected` (this is negation of the previous name, so would require flipping the true/false). Similarly, `buttons_enable` seems to be referring to kick/chip button only, so the variable name can be updated to be more descriptive aswell.
Also, the `not self.buttons_enable` in the if statement seems to be redundant.

Reply:
I changed it so that buttons_should_enable is now no_auto_selected, so it's true if the buttons should be enabled, and false otherwise